### PR TITLE
nodePackages.{triton,manta}: repackage using buildNpmPackage 

### PR DESCRIPTION
**Pull request by teutat3s**
###### Description of changes
- convert `nodePackages.triton` to buildNpmPackage
- convert `nodePackages.manta` to buildNpmPackage
- add version tests for `triton` and `manta`
- renamed `nodePackages.manta` to `node-manta`, because `manta` is already taken

If desired, can be split up into two separate PRs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teutat3s** [comment]
Related https://github.com/NixOS/nixpkgs/issues/229475

File: pkgs/tools/admin/manta/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  inherit (source) version;
+
+  src = fetchurl {
+    url = "https://registry.npmjs.org/${pname}/-/${source.filename}";
+    hash = source.integrity;
+  };
+
+  npmDepsHash = source.deps;
+
+  dontNpmBuild = true;
```
Comment (winterqt) [inline pkgs/tools/admin/manta/default.nix:17]: You can also just set `dontBuild = true`, if you don't need the default Makefile build phase to run.

File: pkgs/tools/admin/manta/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      package = node-manta;
+    };
+    updateScript = ./update.sh;
+  };
+
+  meta = with lib; {
+    description = "Manta Client CLIs and Node.js SDK (https://www.tritondatacenter.com/triton/object-storage)";
+    homepage = "https://github.com/TritonDataCenter/node-manta";
+    license = licenses.mit;
+    platforms = platforms.all;
```
Comment (winterqt) [inline pkgs/tools/admin/manta/default.nix:58]: `buildNpmPackage` sets this by default, you should only set it if you have a good reason to do so. (Plus, `all` wouldn't even be correct here, I don't think.)

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/triton/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}:
+
+buildNpmPackage rec {
+  pname = "triton";
+  version = "7.15.4";
+
+  src = fetchFromGitHub {
+    owner = "TritonDataCenter";
+    repo = "node-triton";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/triton/default.nix:16]:
```suggestion
    rev = version;
```

File: pkgs/tools/admin/triton/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    tests.version = testers.testVersion {
+      package = triton;
+    };
+  };
+
+  meta = with lib; {
+    description = "TritonDataCenter CLI and client (https://www.tritondatacenter.com/triton/compute)";
+    homepage = "https://github.com/TritonDataCenter/node-triton";
+    license = licenses.mpl20;
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/triton/default.nix:42]:
```suggestion
```

buildNpmPackage sets this to nodejs.meta.platforms

File: pkgs/tools/admin/manta/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/admin/manta/default.nix:58]:
```suggestion
```

File: pkgs/tools/admin/manta/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = node-manta;
+    };
+    updateScript = ./update.sh;
+  };
+
+  meta = with lib; {
+    description = "Manta Client CLIs and Node.js SDK (https://www.tritondatacenter.com/triton/object-storage)";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/manta/default.nix:55]:
```suggestion
    description = "Manta Client CLIs and Node.js SDK";
```

File: pkgs/tools/admin/manta/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
+{ lib, buildNpmPackage, fetchurl, nodejs, installShellFiles, testers, node-manta }:
+
+let
+  source = builtins.fromJSON (builtins.readFile ./source.json);
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/manta/default.nix:4]:
```suggestion
  source = lib.importJSON ./source.json;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please rebase and add an alias to the new aliases file inside of node-packages.

File: pkgs/tools/admin/manta/default.nix (hunk 1 above)
Comment (teutat3s) [reply pkgs/tools/admin/manta/default.nix]: Fixed, thanks.

File: pkgs/tools/admin/manta/default.nix (hunk 2 above)
Comment (teutat3s) [reply pkgs/tools/admin/manta/default.nix]: Fixed, thanks.

File: pkgs/tools/admin/triton/default.nix (hunk 3 above)
Comment (teutat3s) [reply pkgs/tools/admin/triton/default.nix]: Fixed, thanks.

File: pkgs/tools/admin/triton/default.nix (hunk 4 above)
Comment (teutat3s) [reply pkgs/tools/admin/triton/default.nix]: Fixed, thanks.

File: pkgs/tools/admin/manta/default.nix (hunk 5 above)
Comment (teutat3s) [reply pkgs/tools/admin/manta/default.nix]: Fixed, thanks.

File: pkgs/tools/admin/manta/default.nix (hunk 6 above)
Comment (teutat3s) [reply pkgs/tools/admin/manta/default.nix]: Fixed, thanks.

File: pkgs/development/node-packages/aliases.nix [hunk 7]
```diff
@@ -35,4 +35,6 @@ in
 
 mapAliases ({
   "@githubnext/github-copilot-cli" = pkgs.github-copilot-cli; # Added 2023-05-02
+  "manta" = pkgs.node-manta; # Added 2023-05-06
+  "triton" = pkgs.triton; # Added 2023-05-06
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/aliases.nix:39]:
```suggestion
  manta = pkgs.node-manta; # Added 2023-05-06
  triton = pkgs.triton; # Added 2023-05-06
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/node-packages/aliases.nix (hunk 7 above)
Comment (teutat3s) [reply pkgs/development/node-packages/aliases.nix]: Thanks for taking care of this.

**Comment by SuperSandro2000** [comment]
@ofborg build triton node-manta

**Comment by dotlambda** [comment]
@teutat3s Can you make an upstream PR committing `package-lock.json` to their repo so we don't have to vendor 3k lines?

**Pull request merged**